src/sql/src/plan/expr.rs

```diff
@@ -1314,7 +1314,11 @@ impl HirScalarExpr {
                 *e = params[*i - 1].clone();
                 // Correct any column references in the parameter expression for
                 // its new depth.
-                e.visit_columns(0, &mut |_, col| col.level += depth);
+                e.visit_columns(0, &mut |d, col| {
+                    if col.level >= d {
```

Comment:
Can you add a comment either here or in the commit message detailing the bug, and why this fixes it?

Reply:
yeah. I will use the comment from the PR in the commit message.